### ISSUE TITLE
Grains set in minion_opts do not appear in a call to `grains.items`.

### ISSUE THREAD

**Issue by pirogoeth** [saltstack/salt#27290]
Grains set in my roster file do not appear when I call `grains.items` via salt-ssh. 

Roster file:
```
omega.maio.me:
  host: omega.maio.me
  user: local
  port: 22
  sudo: True
  minion_opts:
    grains:
      roles:
        - nginx
        - docker
        - dokku
      minimal: True
      deployment: baremetal
```

Saltfile:
```
salt-ssh:
  max_procs: 30
  wipe_ssh: True
  roster_file: .salt-secret/roster
  config_dir: .salt-secret/conf
  ssh_priv: .salt-secret/pki/salt_rsa
```

.salt-secret/conf/master:
```
pki_dir: "/home/pirogoeth/maiome-saltstates/.salt-secret/local/pki"
cachedir: "/home/pirogoeth/maiome-saltstates/.salt-secret/local/cache"
log_file: "/home/pirogoeth/maiome-saltstates/.salt-secret/local/log/master"

jinja_trim_blocks: True
jinja_lstrip_blocks: True

state_verbose: True
state_output: full
state_tabular: True

ssh_minion_opts:
  grains:
    transport: ssh

file_roots:
  base:
    - base
    - base/states
    - base/files
  staging:
    - staging
    - staging/states
    - staging/files
  prod:
    - prod
    - prod/states
    - prod/files

pillar_roots:
  base:
    - base/pillars
  staging:
    - staging/pillars
  prod:
    - prod/pillars
```

`salt-ssh 'omega.maio.me' grains.items`:
```
omega.maio.me:
    ----------
    SSDs:
    biosreleasedate:
        10/05/2011
    biosversion:
        F.1B
    cpu_flags:
        - fpu
        - vme
        - de
        - pse
        - tsc
        - msr
        - pae
        - mce
        - cx8
        - apic
        - sep
        - mtrr
        - pge
        - mca
        - cmov
        - pat
        - pse36
        - clflush
        - dts
        - acpi
        - mmx
        - fxsr
        - sse
        - sse2
        - ss
        - ht
        - tm
        - pbe
        - syscall
        - nx
        - rdtscp
        - lm
        - constant_tsc
        - arch_perfmon
        - pebs
        - bts
        - rep_good
        - nopl
        - xtopology
        - nonstop_tsc
        - aperfmperf
        - eagerfpu
        - pni
        - pclmulqdq
        - dtes64
        - monitor
        - ds_cpl
        - vmx
        - est
        - tm2
        - ssse3
        - cx16
        - xtpr
        - pdcm
        - pcid
        - sse4_1
        - sse4_2
        - x2apic
        - popcnt
        - tsc_deadline_timer
        - aes
        - xsave
        - avx
        - lahf_lm
        - ida
        - arat
        - epb
        - xsaveopt
        - pln
        - pts
        - dtherm
        - tpr_shadow
        - vnmi
        - flexpriority
        - ept
        - vpid
    cpu_model:
        Intel(R) Core(TM) i5-2410M CPU @ 2.30GHz
    cpuarch:
        x86_64
    domain:
        internal.maio.me
    fqdn:
        omega.internal.maio.me
    fqdn_ip4:
        - 127.0.1.1
    fqdn_ip6:
    gpus:
        |_
          ----------
          model:
              2nd Generation Core Processor Family Integrated Graphics Controller
          vendor:
              intel
    host:
        omega
    hwaddr_interfaces:
        ----------
        em1:
            2c:27:d7:b1:ed:76
        lo:
            00:00:00:00:00:00
        wlan0:
            ac:81:12:4f:a7:b1
    id:
        omega.internal.maio.me
    init:
        upstart
    ip4_interfaces:
        ----------
        em1:
            - 10.0.1.18
        lo:
            - 127.0.0.1
        wlan0:
    ip6_interfaces:
        ----------
        em1:
            - snip
            - snip
            - fe80::2e27:d7ff:feb1:ed76
        lo:
            - ::1
        wlan0:
    ip_interfaces:
        ----------
        em1:
            - 10.0.1.18
            - snip
            - snip
            - fe80::2e27:d7ff:feb1:ed76
        lo:
            - 127.0.0.1
            - ::1
        wlan0:
    ipv4:
        - 10.0.1.18
        - 127.0.0.1
    ipv6:
        - ::1
        - snip
        - snip
        - fe80::2e27:d7ff:feb1:ed76
    kernel:
        Linux
    kernelrelease:
        3.13.0-24-generic
    locale_info:
        ----------
        defaultencoding:
            UTF-8
        defaultlanguage:
            en_US
        detectedencoding:
            UTF-8
    localhost:
        omega.internal.maio.me
    lsb_distrib_codename:
        trusty
    lsb_distrib_description:
        Ubuntu 14.04 LTS
    lsb_distrib_id:
        Ubuntu
    lsb_distrib_release:
        14.04
    machine_id:
        snip
    manufacturer:
        Hewlett-Packard
    master:
        salt
    mdadm:
    mem_total:
        15996
    nodename:
        omega.internal.maio.me
    num_cpus:
        4
    num_gpus:
        1
    os:
        Ubuntu
    os_family:
        Debian
    osarch:
        amd64
    oscodename:
        trusty
    osfinger:
        Ubuntu-14.04
    osfullname:
        Ubuntu
    osrelease:
        14.04
    osrelease_info:
        - 14
        - 4
    path:
        /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/bin:/usr/local/sbin
    productname:
        HP Pavilion dv6 Notebook PC
    ps:
        ps -efHww
    pythonexecutable:
        /usr/bin/python2.7
    pythonpath:
        - /tmp/.local_4f7b34_salt
        - /usr/lib/python2.7
        - /usr/lib/python2.7/plat-x86_64-linux-gnu
        - /usr/lib/python2.7/lib-tk
        - /usr/lib/python2.7/lib-old
        - /usr/lib/python2.7/lib-dynload
        - /usr/local/lib/python2.7/dist-packages
        - /usr/lib/python2.7/dist-packages
    pythonversion:
        - 2
        - 7
        - 6
        - final
        - 0
    saltpath:
        /tmp/.local_4f7b34_salt/salt
    saltversion:
        2015.8.0
    saltversioninfo:
        - 2015
        - 8
        - 0
        - 0
    serial:
        snip
    server_id:
        snip
    shell:
        /bin/bash
    uuid:
        snip
    virtual:
        physical
```

**Comment by pirogoeth** [comment]
@basepi ^^

**Comment by jfindlay** [comment]
@pirogoeth, thanks for the report.

**Comment by basepi** [comment]
Fixed in #27573.

**Comment by pirogoeth** [comment]
@basepi Thanks!
